dts/arm/silabs/efr32bg22-pinctrl.dtsi

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright (c) 2022 Silicon Labs
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <dt-bindings/pinctrl/gecko-pinctrl.h>
+
+&pinctrl {
+	/* configuration for uart0 device, default state */
+	usart0_default: usart0_default {
+		group1 {
+			/* configure PE.11 as UART_RX and PE.10 as UART_TX */
+			psels = <GECKO_PSEL(UART_TX, GECKO_PORT_E, GECKO_PIN(10))>,
+				<GECKO_PSEL(UART_RX, GECKO_PORT_E, GECKO_PIN(11))>,
+				<GECKO_LOC(UART_LOC, GECKO_LOCATION(0))>;
```

Comment:
suggestion: to make code less verbose, you can concatenate (`GECKO_ ##`) in GECKO_PSEL and have something like:

`<GECKO_PSEL(UART_TX, PORT_E, PIN(10))>`

Reply:
I see your point and I fully agree but this would require changes across multiple SiLabs SoCs. I think it would be best to do that in a separate PR.

Reply:
I'm sorry, you are right, I was looking at `GECKO_LOCATION` for some reason - this macro is used in all other SiLabs SoCs while `GECKO_PSEL` is used only in EFR32BG22. I added the concatenations to the macro definition as you requested, edited relevant part of bindings and modified all usages to take the advantage of shorter args.